src/Generator/Generators/ExtensionMethods.cs

```diff
@@ -23,5 +24,30 @@ public static Interop.CallingConvention ToInteropCallConv(this CallingConvention
 
             return Interop.CallingConvention.Winapi;
         }
+
+        public static bool CheckClassIsStructible(Class @class, Driver Driver)
+        {
+            if (!@class.IsStruct)
+                return false;
+            if (@class.IsInterface || @class.IsStatic || @class.IsAbstract)
+                return false;
+            if (@class.Methods.Any(m => m.IsOperator))
+                return false;
+
+            if (@class.IsValueType)
```

Comment:
Yes, I actually have changed many things from this pull but in a separate branch. I will do that here as well.